### ISSUE TITLE
Example Suggestions: Heat quation, Lorenz equations, Lotka-Volterra equations and Sturm-Liouville Equation

### ISSUE THREAD

**Issue by fusion809** [davidrpugh/pyCollocation#46]
Hi,

I'm quite the fan of this Python module, and I wanted to suggest the following examples:
- The [Heat equation](https://en.wikipedia.org/wiki/Heat_equation), that is:

![](https://wikimedia.org/api/rest_v1/media/math/render/svg/6544d3fe02c45c7f36beb824d68073274d7b60b3)
- The [Lorenz equations](https://en.wikipedia.org/wiki/Lorenz_system):

![](https://wikimedia.org/api/rest_v1/media/math/render/svg/5f993e17e16f1c3ea4ad7031353c61164a226bb8)

Would be best to solve it with the original parameters: ![sigma](https://wikimedia.org/api/rest_v1/media/math/render/svg/f3e7e6795cde44bcd0913e18917d7de69575cca5), ![beta](https://wikimedia.org/api/rest_v1/media/math/render/svg/7639fbf77fddf40dc63b185e40a4676ac55d30c5) and ![rho](https://wikimedia.org/api/rest_v1/media/math/render/svg/e3a9e7187d23db35f1c1c1e80d3ce289d0d3092d). `x[0] = y[0] = z[0] = 1` on t &#8712; [0,100]. 
- The [Lotka-Volterra equations](https://en.wikipedia.org/wiki/Lotka%E2%80%93Volterra_equations), that is:

![](https://wikimedia.org/api/rest_v1/media/math/render/svg/79752d662d4760abcc84c6f0bb94d708f17ff442)
- The [Sturm-Liouville Equation](https://en.wikipedia.org/wiki/Sturm%E2%80%93Liouville_theory), specifically the example:

![](https://wikimedia.org/api/rest_v1/media/math/render/svg/2e036052d7599dbf0a33db64d401039096a022c3)

where lambda is the eigenvalue of the equation. This problem is integrated over the range x &#8712; [0,&infin;], although a range of `[0,100]` is good enough to get the first 10 or more eigenvalues. Plus it has the boundary conditions y(0) = y(&infin;) = 0. This equation has the exact solution:

![](https://wikimedia.org/api/rest_v1/media/math/render/svg/7d07e6b5511e18e50b805c790f3ff82ebb1e5bc8)

where Ai is the [Airy Ai function](https://en.wikipedia.org/wiki/Airy_function). The eigenvalues are the zeros of the Airy Ai function, with their negative sign replaced with a positive sign.

**Comment by davidrpugh** [comment]
@fusion809 Thanks! Always good to hear from a user.  I will look into adding these as examples.  Please fell free to submit a PR for you favorite examples and I will certainly include it!

**Comment by fusion809** [comment]
I don't think this module can presently be used to solve Sturm-Liouville problems, can it?

**Comment by davidrpugh** [comment]
I do know.  I have never solved a Sturm-Liouville problem before.  Can you explain what is required?

**Comment by fusion809** [comment]
Well, I must admit part of the reason I want more examples is because I am so unfamiliar with your present examples that I can't gather how to use pyCollocation from them. So I was hoping examples I'm more familiar with might help me follow it, although it does like quite a helpful tool. Essentially what you do is you express the problem in terms of matrices, e.g., say the `H` matrix represents LHS of the above SL problem then you can express the problem as the matrix equation:

```
Hy = lambda y
```

then it becomes a eigenvalue/eigenvector problem from linear algebra, with the solution contained in the `y` vector being the eigenvector and `lambda` being the eigenvalues. In terms of what you'd use, Python module-wise, it'd probably be [linalg.eig](http://docs.scipy.org/doc/numpy/reference/generated/numpy.linalg.eig.html).

**Comment by davidrpugh** [comment]
PyCollocation was designed to solve non-linear boundary problems commonly encountered in economics. Which might be why the examples seem so unfamiliar.  Currently you need to be able to express your problem as `y` = f(t, y)` along with some boundary conditions. Is the SL problem always linear?

**Comment by fusion809** [comment]
Yep they are always linear, their most generalized form is:

![](https://wikimedia.org/api/rest_v1/media/math/render/svg/7b80b2cfbaabababb3a0f02a570c4a705d7e748b)

Most notable one is the Schrodinger equation. Although nonlinear ODE eigenvalue problems do exist.

**Comment by davidrpugh** [comment]
OK.  I have never tried to solve a linear problem using pyCollocation.  I am pretty busy at the moment, but when I get some free time I will see if it is possible.  I am pretty sure it is possible...
